### DISCUSSION TITLE
没考虑用hook重写吗？替换class api

### DISCUSSION THREAD

**Discussion by gitHber** [DouyinFE/semi-design#985] [Q&A] [answered]
有计划重构成hook版本吗？像form 获取实例之类的还在使用 ref 这种是不是有点过时

**Comment by pointhalo** [comment, marked as answer]
先说下结论：【不计划】
再说下我们的看法：
hooks与class各有好处，不存在谁比谁一定更优这种说法，不应该脱离使用场景来评价。
1. 在写业务组件时，hooks会更便捷一些。
在写基础UI组件时，因为有很多特定交互逻辑是依赖生命周期函数的，所以传统的class反而更有优势。而且对于复杂组件而言，hooks的形式组织代码，在可阅读性上跟class会有一定差距。

2. Semi 的组件架构是 foundation / adapter 分层的。foundation 层完全与前端框架无关，可用于适配任意前端技术栈。hooks 仅限 react 体系内，不好支持我们做与框架无关的逻辑抽象。

**Comment by pointhalo** [comment]
这个话题更适合作为 discussion，我转一下哈

**Reply by gitHber** [reply]
见仁见智吧，目前vue 和 react都转hook， 其实class 和 hook 还是存在一定的不兼容的，从类型还是使用方法都有点差别，而且像 form 表单这种获取实例通过 hook 形式肯定是更方便的

**Reply by pointhalo** [reply]
Semi 的长期目标，也是采用 F/A 架构的初衷，会向 WebComponent 靠拢，所以不是很倾向于以移植性低的形式去组织组件的逻辑。https://bytedance.feishu.cn/wiki/wikcnOVYexosCS1Rmvb5qCsWT1f

至于单就form这一组件而言，ref也并非是获取的唯一方式，也提供了多种 get method
